File: src/Dir/Dir_File_Scan.php
<?php

namespace xltxlm\fileinfo\Dir;

use RecursiveDirectoryIterator;

/**
 * 扫描文件夹的文件.;
 */
class Dir_File_Scan extends Dir_File_Scan\Dir_File_Scan_implements
{

    /**
     * Dir_File_Scan constructor.
     */
    public function __construct(string $dir = '')
    {
        if ($dir) {
            $this->setdir($dir);
        }
    }

    /**
     * @return \SplFileInfo[];
     */
    public function __invoke()
    {
        if (!$this->dir) {
            return [];
        }
        $Directory = (new RecursiveDirectoryIterator($this->getdir()));
        $Iterator = new \RecursiveIteratorIterator($Directory);
        if ($this->depth !== null) {
            $Iterator->setMaxDepth($this->getDepth());
        }
        /** @var \SplFileInfo $item */
        foreach ($Iterator as $item) {
            if ($item->getFilename() == '..' || $item->getRealPath() == $this->getDir()) {
                continue;
            }

            if ($item->isDir()) {
                continue;
            }
            if ($this->getPreg()) {
                $match = preg_match('#' . $this->getPreg() . '#i', basename($item->getRealPath()));
                if (!$match) {
                    continue;
                }
            }
            yield $item;
        }
    }

}

File: go/Dir/Dir_File_Scan.go
package Dir

type Dir_File_Scan struct {
    /* 只有匹配的才返回,正则表达式 */
    preg string

    /* 文件夹深度 */
    depth int

    /* 要检索的文件夹名称 */
    dir string

}

func NewDir_File_Scan() *Dir_File_Scan{
    var this = new(Dir_File_Scan)
    return this
}


/**
    */
func (this *Dir_File_Scan)__invoke(){

}

